capstone-project/src/main/webapp/map/map.component.js

```diff
@@ -7,25 +7,88 @@ angular.module('map').component('mapComponent', {
     templateUrl: 'map/map.template.html',
     controller: function($scope, $http) {
 
-        let googleMapOption = {
+        // Define the map.
+        const googleMapOption = {
             zoom: 4,
             center: new google.maps.LatLng(25, 80)
         };
-
         $scope.gMap = new google.maps.Map(document.getElementById('map-container'), googleMapOption);
 
-        // Add a marker to the map
-        const addMarker = function(marker) {
+        // When the user clicks on the map, show a marker with a text box the user can edit.
+        $scope.gMap.addListener('click', (event) => {
+            addMarkerForEdit(event.latLng.lat(), event.latLng.lng());
+        });
+
+        // Editable marker that displays when a user clicks on the map. 
+        let editableMarker;
+        // Add a marker the user can edit. 
+        const addMarkerForEdit = function(lat, lng) {
+            // If we're already showing an editable marker, then remove it.
+            if (editableMarker) {
+                editableMarker.setMap(null);
+            }
+            editableMarker = new google.maps.Marker({ position: { lat: lat, lng: lng }, map: $scope.gMap });
+            const infoWindow = new google.maps.InfoWindow({ content: buildInfoWindowInput(lat, lng) });
+
+            // When the user closes the editable info window, remove the marker.
+            google.maps.event.addListener(infoWindow, 'closeclick', () => {
+                editableMarker.setMap(null);
+            });
+
+            infoWindow.open($scope.gMap, editableMarker);
+        }
+
+        // Build and return HTML elements that show editable textboxes and a submit button.
+        const buildInfoWindowInput = function(lat, lng) {
+            const animal = document.createElement('textarea');
+            animal.placeholder = "Animal";
+            const description = document.createElement('textarea');
+            description.placeholder = "Description";
+            const reporter = document.createElement('textarea');
+            reporter.placeholder = "Reporter's Name";
+            const button = document.createElement('button');
+            button.appendChild(document.createTextNode('Submit'));
 
-            new google.maps.Marker({
+            button.onclick = () => {
+                let newMarker = {
```

Comment:
Can be const.

Make sure let variables are only set for variables that change their value. It makes it easier to read.